emscripten-functions/src/websocket.rs

```diff
@@ -0,0 +1,366 @@
+    pub fn clear_internal_callback(&mut self) {
+        unsafe {
+            emscripten_websocket_set_onopen_callback_on_thread(
+                self.id,
+                std::ptr::null_mut::<std::os::raw::c_void>(),
+                None,
+                EM_CALLBACK_THREAD_CONTEXT_CALLING_THREAD,
+            );
+            emscripten_websocket_set_onerror_callback_on_thread(
+                self.id,
+                std::ptr::null_mut::<std::os::raw::c_void>(),
+                None,
+                EM_CALLBACK_THREAD_CONTEXT_CALLING_THREAD,
+            );
+            emscripten_websocket_set_onclose_callback_on_thread(
+                self.id,
+                std::ptr::null_mut::<std::os::raw::c_void>(),
+                None,
+                EM_CALLBACK_THREAD_CONTEXT_CALLING_THREAD,
+            );
+            emscripten_websocket_set_onmessage_callback_on_thread(
+                self.id,
+                std::ptr::null_mut::<std::os::raw::c_void>(),
+                None,
+                EM_CALLBACK_THREAD_CONTEXT_CALLING_THREAD,
+            );
+        }
+    }
+
+    /// Set on open callback for current object.    
+    pub fn set_open_callback(&mut self, cb: Option<fn(&mut Self)>) {
+        self.open_cb = cb;
+    }
+    /// Set on error callback for current object.    
+    pub fn set_error_callback(&mut self, cb: Option<fn(&mut Self)>) {
+        self.error_cb = cb;
+    }
+    /// Set on close callback for current object.    
+    pub fn set_close_callback(&mut self, cb: Option<fn(&mut Self)>) {
+        self.close_cb = cb;
+    }
+    /// Set on message callback for current object.    
+    pub fn set_message_callback(&mut self, cb: Option<fn(&mut Self, WebSocketData)>) {
+        self.message_cb = cb;
+    }
+
+    /// Send UTF-8 formatted string through websocket. The state of current socket should be
+    /// opened else the function will return false.    
+    ///
+    /// # Examples
+    /// ```rust
+    /// let mut ws = WebSocket::new().unwrap();
+    /// ws.connect("wss://echo.websocket.org/");
+    /// ws.send_utf8_text("foo");
+    /// ```
+    pub fn send_utf8_text(&mut self, str: &str) -> bool {
+        if self.state == WebSocketState::Opened {
+            let text_cstr = CString::new(str).unwrap();
+            unsafe {
+                emscripten_websocket_send_utf8_text(self.id, text_cstr.as_ptr());
+            }
+
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Send UTF-8 raw data through websocket. The state of current socket should be
+    /// opened else the function will return false.
+    ///
+    /// # Examples
+    /// ```rust
+    /// let mut ws = WebSocket::new().unwrap();
+    /// ws.connect("wss://echo.websocket.org/");
+    /// ws.send_binary([42, 69]);
+    /// ```
+    pub fn send_binary(&mut self, data: &mut [u8]) -> bool {
+        if self.state == WebSocketState::Opened {
+            unsafe {
+                emscripten_websocket_send_binary(
```

Comment:
Probably better to just return whether the function's return value is equal to `EMSCRIPTEN_RESULT_SUCCESS`